### PR TITLE
Address NPM security issues in `immer` + `json-schema`

### PR DESCRIPTION
**Pull request by snazy**
* Transitive dependency `immer@8.0.1` via `react-dev-utils@11.0.4`
  CVE-2021-3757 (high), CVE-2021-23436 (moderate)
* Transitive dependency `json-schema@0.3.0` via `workbox-webpack-plugin@6.4.1` via `@apideck/better-ajv-errors@0.2.7`
  CVE-2021-3918 (moderate)

### CODE REVIEW

**Comment by ajantha-bhat** [comment]
Other way to fix immer CVE is by waiting for new release I guess [https://github.com/facebook/create-react-app/issues/11660]

also @dimas-b tried https://github.com/projectnessie/nessie/pull/2689

File: ui/package.json [hunk 1]
```diff
@@ -113,16 +112,19 @@
     "fs-extra": "^10.0.0",
     "html-webpack-plugin": "4.5.2",
     "husky": "^7.0.4",
+    "immer": ">=9.0.6",
```
Comment (dimas-b) [inline ui/package.json:115]: I'm not sure this helps. AFAIK, `npm` will still use embedded `immer` versions declared in transitive dependencies for their respective "client" modules.

File: ui/package.json [hunk 2]
```diff
@@ -113,16 +112,19 @@
...
     "jest-resolve": "27.3.1",
     "jest-watch-typeahead": "1.0.0",
+    "json-schema": ">=0.4.0",
     "lint-staged": "^12.1.2",
     "mini-css-extract-plugin": "1.6.2",
     "nock": "^13.2.1",
     "optimize-css-assets-webpack-plugin": "6.0.1",
     "pnp-webpack-plugin": "1.7.0",
     "prettier": "^2.4.1",
+    "react-dev-utils": "^11.0.4",
```
Comment (dimas-b) [inline ui/package.json:127]: This `react-dev-utils` still references `immer` 8.x AFAIK

**Comment by snazy** [comment]
Opened #2744 to move react-dev-utils out of the prod dependencies, so the security issues shouldn't be that problematic.

**Pull request closed without merging**